### ISSUE TITLE
Namespaces are not respect for local struct variables

### ISSUE THREAD

**Issue by g33kex** [PainsPerdus/gboi-rocket#33]
## Issue 

Namespaces are not respected for structs that should be local in .var files. 
For instance, if the struct Isaac is declared in global.var. s as such : 
~~~nasm
.STRUCT isaac
    x DB
    y DB
    hp DB
    dmg DB
    upgrades DW
    range DB
    speed DB
    tears DB
    recover DB
    bombs DB
.ENDST
~~~

And added to the namespace like so:
~~~nasm
.STRUCT global var
   sheets INSTANCEOF sheet n_sheets
    isaac INSTANCEOF isaac
    elements INSTANCEOF element n_elements
    issac_tear_pointer DB
    isaac_tears INSTANCEOF tear n_isaac_tears
    ennemy_tear_pointer DB
    ennemy_tears INSTANCEOF tear n_ennemy_tears
    states INSTANCEOF state n_states
.ENDST
~~~

This struct can be accessed with `global_.isaac` (this is intended behavior)

Then, another local struct named Isaac can't be declared in another .var file (ex : display.var.s):
~~~nasm
.STRUCT isaac 
    frame DB
    shoot_timer DB
    walk_timer DB
.ENDST
~~~

This results in assembler error: 
```
src/var/display.var.s:5: DIRECTIVE_ERROR: "_sizeof_isaac" was defined for the second time.
```

## Workaround

We need to define the new Isaac as so with a different name:
~~~nasm
.STRUCT new_isaac
    frame DB
    shoot_timer DB
    walk_timer DB
.ENDST
~~~

And then add it to the namespace with the same name:
~~~nasm
.STRUCT display_var
    isaac INSTANCEOF new_isaac
.ENDST
~~~

This works, we can now call global Isaac with `global_.isaac` and display Isaac with `display_.isaac`.

The issue is that people can't necessarily know the name of local struct used in .var files, as such a lot of errors like these might appear.

## Possible (somewhat ugly?) fix

If it's not possible to fix that with the namespace system, I suggest we should use UUID for these local variables, for instance :
~~~nasm
.STRUCT U38LKJA9B9R
   frame DB
   shoot_timer DB
   walk_timer DB
.ENDST

.STRUCT display_var
   isaac INSTANCEOF U38LKJA9B9R
.ENDST
~~~

But this is not very comprehensible/practical. Having to add the namespace to the struct name (ex:  `display_.isaac`) would completely defeat the purpose of the semi-automated namespacing system...

**Comment by JeanMarieMineau** [comment]
Yep I know... I still can't figure out how to make the official namespaces work without a lot of (worse) issues... It's not a big deal because the name of the struct is only used in the .var, so you can use less understandable names (for the struct name, not the instance...). You can prefix them with the name of the module for instance.
The same issue occurs with the defines, and it a lot more problematic. Still working on it.
